pkg/jiraservicedesk/client/client.go

```diff
@@ -73,6 +77,36 @@ func (c *jiraServiceDeskClient) newRequest(method, path string, body interface{}
 	return req, nil
 }
 
+func (c *jiraServiceDeskClient) newRequestWithExperimentalHeader(method, path string, body interface{}) (*http.Request, error) {
+	endpoint := c.baseURL + path
+	url, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var buf io.ReadWriter
+
+	if body != nil {
+		buf = new(bytes.Buffer)
+		err := json.NewEncoder(buf).Encode(body)
+		if err != nil {
+			return nil, err
+		}
+	}
+	req, err := http.NewRequest(method, url.String(), buf)
+	if err != nil {
+		return nil, err
+	}
+	if body != nil {
+		req.Header.Set("Content-Type", "application/json")
+	}
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("User-Agent", "golang httpClient")
+	req.Header.Set("X-ExperimentalApi", "opt-in")
+	req.SetBasicAuth(c.email, c.apiToken)
+	return req, nil
+}
```

Comment:
Why do we need a separate method fro this ? Can't we add a flag in the parameters of the already existing method ? Or maybe simply add this header there as well, without any additional flags/parameters

Reply:
Yes, we can add a parameter into the newRequest() function and provide the argument that can check if the experimental header should be added in the function or not. But the function newRequest was being used at many different places both in the customer and the project. So, I thought maybe if it is a better approach to write a new function for an api call with an experimental header rather than updating the newRequest() function argument list at every other place.

Reply:
Just add this header in there without any additional arguments. And then update the documentation, in the customers section do add that the api we are using is in experimental stage at the moment. Also, please mention other limitations like we cannot delete a user there as well

Reply:
Resolved the issue by updating the adding an additional argument i.e. Experimental in the newRequest method. Removed the newRequestWithExperimentalHeader method.